src/Language/PureScript/Make.hs

```diff
@@ -136,7 +136,7 @@ rebuildModuleWithIndex MakeActions{..} exEnv externs m@(Module _ _ moduleName _
 --
 -- If timestamps or hashes have not changed, existing externs files can be used to provide upstream modules' types without
 -- having to typecheck those modules again.
-make :: forall m. (Monad m, MonadBaseControl IO m, MonadError MultipleErrors m, MonadWriter MultipleErrors m)
+make :: forall m. (Monad m, MonadBaseControl IO m, MonadError MultipleErrors m, MonadWriter MultipleErrors m, MonadIO m)
```

Comment:
What is this `MonadIO` constraint added for?

Reply:
I needed to check which files existed in order to remove them from the cache which meant I had to do IO. I'm kind of new to haskell but my LSP suggested i add that constraint and it just worked. It felt kind of like it shouldn't be there when I wrote it, not sure how I would do it otherwise

Reply:
Right, I just couldn't see what required this constraint straight away